src/ResourceManager/DataLakeAnalytics/Commands.DataLakeAnalytics/Commands/SetAzureRmDataLakeAnalyticsCatalogCredential.cs

```diff
@@ -0,0 +1,105 @@
+            Mandatory = true, HelpMessage = "The name of the database to create the credential in.")]
+        [Parameter(ValueFromPipelineByPropertyName = true, ParameterSetName = HostAndPortParameterSetName, Position = 1,
+            Mandatory = true, HelpMessage = "The name of the database to create the credential in.")]
+        [ValidateNotNullOrEmpty]
+        public string DatabaseName { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, ParameterSetName = BaseParameterSetName, Position = 1,
+            Mandatory = true, HelpMessage = "The name of the credential to create.")]
+        [Parameter(ValueFromPipelineByPropertyName = true, ParameterSetName = HostAndPortParameterSetName, Position = 1,
+            Mandatory = true, HelpMessage = "The name of the credential to create.")]
+        [ValidateNotNullOrEmpty]
+        public string CredentialName { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, ParameterSetName = BaseParameterSetName, Position = 2,
+            Mandatory = true, HelpMessage = "The credential to create, which includes the user ID and password that can authenticate to the data source")]
+        [Parameter(ValueFromPipelineByPropertyName = true, ParameterSetName = HostAndPortParameterSetName, Position = 2,
+            Mandatory = true, HelpMessage = "The credential to create, which includes the user ID and password that can authenticate to the data source")]
+        [ValidateNotNullOrEmpty]
+        public PSCredential Credential { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, ParameterSetName = BaseParameterSetName, Position = 3,
+            Mandatory = true, HelpMessage = "The new password for the credential")]
+        [Parameter(ValueFromPipelineByPropertyName = true, ParameterSetName = HostAndPortParameterSetName, Position = 3,
+            Mandatory = true, HelpMessage = "The new password for the credential")]
+        [ValidateNotNullOrEmpty]
+        public PSCredential NewPassword { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, ParameterSetName = HostAndPortParameterSetName, Position = 4,
+            Mandatory = true, HelpMessage = "The URI of the database to connect to.")]
+        public Uri Uri { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, ParameterSetName = BaseParameterSetName, Position = 4,
+            Mandatory = true, HelpMessage = "The host of the database to connect to in the format 'myhost.dns.com'.")]
+        public string DatabaseHost { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, ParameterSetName = BaseParameterSetName, Position = 5,
+            Mandatory = true, HelpMessage = "The Port associated with the host for the database to connect to.")]
+        public int Port { get; set; }
+
+        [Parameter(Position = 6, Mandatory = false, HelpMessage = "Do not ask for confirmation.")]
```

Comment:
switch parameters should not be positional, also 5 positional parameters borders on too many. 

Also, Force is unnecessary here, there are no special conditions you are prompting the user for.  I would remove this parameter.